ui/app/src/components/RenameAssetDialog/RenameAssetDialogContainer.tsx

```diff
@@ -0,0 +1,231 @@
+  const newAssetPath = `${getParentPath(path)}/${newAssetName}`;
+  const assetExists = newAssetName !== value && itemLookupTable[newAssetPath] !== UNDEFINED;
+  const isValid = !isBlank(name) && !assetExists && name !== value;
+  const siteId = useActiveSiteId();
+  const [confirm, setConfirm] = useState(null);
+  const [confirmBrokenReferences, setConfirmBrokenReferences] = useState(false);
+  const { formatMessage } = useIntl();
+  const renameDisabled = isSubmitting || !isValid || (dependantItems.length > 0 && !confirmBrokenReferences);
+  const authoringBase = useSelection<string>((state) => state.env.authoringBase);
+  const [contextMenu, setContextMenu] = useSpreadState({
+    el: null,
+    dependency: null
+  });
+
+  const onInputChanges = (newValue: string) => {
+    setName(newValue);
+    const newHasPendingChanges = newValue !== value;
+    hasPendingChanges !== newHasPendingChanges &&
+      dispatch(updateRenameAssetDialog({ hasPendingChanges: newHasPendingChanges }));
+  };
+
+  const onRenameAsset = (siteId: string, path: string, name: string) => {
+    const fileName = type !== 'asset' ? getName(type, name) : name;
+    renameContent(siteId, path, `${fileName}`).subscribe({
+      next() {
+        onRenamed?.({ path, name });
+        dispatch(updateRenameAssetDialog({ isSubmitting: false, hasPendingChanges: false }));
+      },
+      error(response) {
+        dispatch(showErrorDialog({ error: response }));
+      }
+    });
+  };
+
+  const onConfirmCancel = () => {
+    setConfirm(null);
+    dispatch(updateRenameAssetDialog({ isSubmitting: false }));
+  };
+
+  const onRename = () => {
+    dispatch(updateRenameAssetDialog({ isSubmitting: true }));
+    if (name) {
+      validateActionPolicy(siteId, {
+        type: 'RENAME',
+        target: newAssetPath
+      }).subscribe(({ allowed, modifiedValue }) => {
+        if (allowed && modifiedValue) {
+          setConfirm({
+            body: formatMessage(translations.createPolicy, { name: modifiedValue.replace(`${path}/`, '') })
+          });
+        } else if (allowed) {
+          onRenameAsset(siteId, path, name);
+        } else {
+          setConfirm({
+            error: true,
+            body: formatMessage(translations.policyError)
+          });
+        }
+      });
+    }
+  };
+
+  const handleContextMenuClick = (event: React.MouseEvent<HTMLButtonElement>, dependency: DetailedItem) => {
+    setContextMenu({
+      el: event.currentTarget,
+      dependency
+    });
+  };
+
+  const handleContextMenuClose = () => {
+    setContextMenu({
+      el: null,
+      dependency: null
+    });
+  };
+
+  const handleEditorDisplay = (item: DetailedItem) => {
+    editorDisplay(item, authoringBase, siteId, dispatch, fetchRenameAssetDependants());
+  };
+
+  return (
+    <>
+      <DialogBody>
+        <form
+          onSubmit={(e) => {
+            e.preventDefault();
+            if (isValid) {
```

Comment:
Why is this different than the submit button condition?
```suggestion
            if (renameDisabled) {
```